dna/indexcontent/indexcontent.js

```diff
@@ -0,0 +1,205 @@
+function genesis(){
+
+  //Calling addAnchor function for creating the base anchor
+  baseAnchorHash = call("anchor","addAnchor","");
+  debug("Base anchor added with hash - "+baseAnchorHash);
+
+  var ContentToIndex1 = {content:"holodex : We are Indexing this content using holodex app. this",details:"can include timestamp, etc."};
+  ContentToIndexhash1 = makeHash(ContentToIndex1);
+
+  var ContentToIndex2 = {content:"holodex can also be used for searching keywords",details:"can include timestamp,lication, etc."};
+  ContentToIndexhash2 = makeHash(ContentToIndex2);
+
+
+  //called in genesis temporarily. Once bridging between apps gets workig, index content will be called from the HC app that is
+  //using holodex
+  IndexContent(ContentToIndex1.content,ContentToIndexhash1,"English");
+  IndexContent(ContentToIndex2.content,ContentToIndexhash2,"English");
+
+}
+
+//Function called by the HC app to search for a string of words and get all the objects indexed for the words.
+function searchKeywords(searchString)
+{
+  var searchArr = searchString.split(/,| |:|-/);
+  var i = searchArr.length;
+  var mergedList = [];
+  var list = [];
+
+  debug("Array contents : ");
+  i--;
+
+  while(i>=0)
+  {
+    debug("While loop ------------ i = "+i);
+    debug(searchArr[i]);
+    list = call("anchor","anchor_list",searchArr[i]);
+
+    var temp = JSON.parse(list);
+    debug(list);
+    for(var m=0;m<temp.length;m++)
+    {
+      var temp1 = JSON.parse(temp[m].Anchor_Text);
+      debug(temp1.Anchor_Text);
+      mergedList=union(mergedList,temp1.Anchor_Text);
+    }
+
+    i--;
+  }
+  var jsonmer = JSON.parse(JSON.stringify(mergedList));
+
+  return jsonmer;
+}
+
+//To provide a list of all the objects that are indexed against the search string
+function union(mergedList,list)
```

Comment:
A much faster way to implement this, is to simply store both lists as keys into a hash table with a simple of true (kind of like the ignorewords list) and then return a list of the keys.  The way you did it with a find is means you scan the list for each key, which means that there n^2 operations.  whereas if you just store all the keys, theres just 3n operations.